internal/se/core/tag.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2025 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package secore
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"fmt"
+)
+
+// GenerateEqualityTag creates a deterministic search tag for equality queries
+//
+// SECURITY NOTE: This function generates the SAME tag for the same value across
+// ALL documents in the collection. This enables efficient equality search but
+// reveals when multiple documents share the same field value (frequency analysis).
+// For fields with low cardinality (e.g., boolean, status codes), consider the
+// privacy implications.
+//
+// The tag is computed as: HMAC-SHA256(key, "eq:collectionID:fieldName" || value)[:16]
+func GenerateEqualityTag(
+	key []byte,
+	collectionID string,
+	fieldName string,
+	value []byte,
+) ([]byte, error) {
```

Comment:
todo: There is never an error returned from this function, please remove the `error` return param - you are adding complexity to everywhere that calls this function.